### PR TITLE
chore: Remove deprecated react-router [WEB-580]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

After reviewing a few options for removing the old router, we've landed on this as the preferred one.

- Router created using `createBrowserRouter`, contains final error boundary and <App/>
- Tests use <BrowserRouter>
- Tests which depend on routes code use a mock to avoid loading the real router and entire <App/>

## Test Plan

- react tests pass
- Clicking from one page to another works successfully
- After logging out, /det/workspaces and other links redirect to /det/login successfully
- [todo / possibly want to mock-test on OSS] If an auth error is passed to handleError, it's re-thrown and the page redirects to /det/logout

## Checklist

- [ ] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/CheckpointModalTrigger.test.tsx [hunk 1]
```diff
@@ -20,6 +19,12 @@ jest.mock('services/api', () => ({
   },
 }));
 
+jest.mock('router', () => ({
```
Comment (ashtonG) [inline webui/react/src/components/CheckpointModalTrigger.test.tsx:22]: please move duplicate mock logic like this to a mock module instead.

File: webui/react/src/router.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  componentDidCatch(): void {
+    // You can also log the error to an error reporting service
+    // logErrorToMyService(error, errorInfo);
+    // console.log('error: ' + error);
+    // console.log('errorInfo: ' + JSON.stringify(errorInfo));
+    // console.log('componentStack: ' + errorInfo.componentStack);
+    router.navigate(paths.logout());
+  }
```
[the remark this answers is not in the document]
Comment (mapmeld) [reply webui/react/src/router.tsx]: ok I can use `isAuthFailure` as was originally in handleError

File: webui/react/src/components/NotesCard.tsx [hunk 3]
```diff
@@ -54,6 +51,7 @@ const NotesCard: React.FC<Props> = ({
     setEditedNotes(existingNotes.current);
   }, [noteChangeSignal]);
 
+  /*
```
[the remark this answers is not in the document]
Comment (keita-determined) [reply webui/react/src/components/NotesCard.tsx]: @mapmeld will we use `useBlocker`? If not, i think we should keep the `useEffect` block

[the remark this answers is not in the document]
Comment (mapmeld) [reply webui/react/src/components/NotesCard.tsx]: `history.block` is no longer available, so I've been looking up how to use `unstable_useBlocker` or `unstable_usePrompt here`

File: webui/react/src/shared/utils/routes.ts [hunk 4]
```diff
@@ -37,17 +30,21 @@ export const isMouseEvent = (ev: AnyMouseEvent | React.KeyboardEvent): ev is Any
...
-};
+
 export const ensureAbsolutePath = (url: string): string => (isAbsolutePath(url) ? url : '/' + url);
 export const filterOutLoginLocation = (location: {
   pathname: string;
 }): { pathname: string } | undefined => {
   return location.pathname.includes('login') ? undefined : clone(location);
 };
+
+export const isFullPath = (url: string): boolean => {
```
Comment (ashtonG) [inline webui/react/src/shared/utils/routes.ts:41]: do we need to dupe this function here and in `routes/utils`?

**Review by ashtonG — APPROVED** [review]
lookin' good! just had a few small questions.

File: webui/react/src/shared/utils/routes.ts (hunk 4 above)
Comment (mapmeld) [reply webui/react/src/shared/utils/routes.ts]: originally I was worried about which files imported what

gave this another round of cleanup, put `isFullPath` back in its original place

**Pull request merged**